sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeterProvider.java

```diff
@@ -5,69 +5,22 @@
 
 package io.opentelemetry.sdk.metrics;
 
-import io.opentelemetry.api.OpenTelemetry;
-import io.opentelemetry.api.metrics.MeterBuilder;
 import io.opentelemetry.api.metrics.MeterProvider;
-import io.opentelemetry.sdk.common.Clock;
-import io.opentelemetry.sdk.internal.ComponentRegistry;
-import io.opentelemetry.sdk.metrics.data.MetricData;
-import io.opentelemetry.sdk.metrics.exemplar.ExemplarSampler;
-import io.opentelemetry.sdk.metrics.export.MetricProducer;
-import io.opentelemetry.sdk.metrics.internal.state.MeterProviderSharedState;
-import io.opentelemetry.sdk.metrics.internal.view.ViewRegistry;
-import io.opentelemetry.sdk.resources.Resource;
-import java.util.ArrayList;
-import java.util.Collection;
-import java.util.Collections;
-import java.util.List;
-import java.util.logging.Logger;
-import javax.annotation.Nullable;
+import io.opentelemetry.sdk.common.CompletableResultCode;
 
 /**
- * {@code SdkMeterProvider} implementation for {@link MeterProvider}.
+ * {@code SdkMeterProvider} provides SDK extensions for {@link MeterProvider}.
  *
  * <p>This class is not intended to be used in application code and it is used only by {@link
- * OpenTelemetry}.
- *
- * <p>WARNING: A MetricProducer is stateful. It will only return changes since the last time it was
- * accessed. This means that if more than one {@link
- * io.opentelemetry.sdk.metrics.export.MetricExporter} has a handle to this MetricProducer, the two
- * exporters will not receive copies of the same metric data to export.
+ * io.opentelemetry.api.OpenTelemetry}.
  */
-public final class SdkMeterProvider implements MeterProvider, MetricProducer {
-
-  private static final Logger LOGGER = Logger.getLogger(SdkMeterProvider.class.getName());
-  static final String DEFAULT_METER_NAME = "unknown";
-  private final ComponentRegistry<SdkMeter> registry;
-  private final MeterProviderSharedState sharedState;
-
-  SdkMeterProvider(
-      Clock clock, Resource resource, ViewRegistry viewRegistry, ExemplarSampler exemplarSampler) {
-    this.sharedState =
-        MeterProviderSharedState.create(clock, resource, viewRegistry, exemplarSampler);
-    this.registry =
-        new ComponentRegistry<>(
-            instrumentationLibraryInfo -> new SdkMeter(sharedState, instrumentationLibraryInfo));
-  }
+public interface SdkMeterProvider extends MeterProvider {
```

Comment:
nit: with the change to an interface, the `builder` method doesn't need `public` on it any more.

Reply:
doh!  I'll learn java one of these days.